### ISSUE TITLE
[Enhancement]: Bind value must be active only after edit 

### ISSUE THREAD

**Issue by prapullac** [appsmithorg/appsmith#20403]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

 Clicking on edit option of the manage environment the bind values button  must be active only after the data is been added or there is an edit been done 

### Steps To Reproduce

1) Navigate to any datasource
2) Add necessary details 
3) Save the datasource
4) Click on Manage Environment button 
 and observe the "Bind Value" button is active 
Scenario 2 : Edit or update the Manage environment
1) Click on Edit option of saved Datasource
2) Navigate to Manage Environment button 
 and observe the "Bind Value" button is active even when the user has not made any changes

### Version

DP

**Comment by btsgh** [comment]
Do ping once the changes are merged into the env @ayushpahwa
